modules/feeds_migrate_ui/src/Form/MigrationForm.php

```diff
@@ -37,96 +29,61 @@ class MigrationForm extends EntityForm {
   protected $migrationPluginManager;
 
   /**
-   * The form factory.
-   *
-   * @var \Drupal\feeds_migrate\Plugin\PluginFormFactory
-   */
-  protected $formFactory;
-
-  /**
-   * Fill This.
-   *
-   * @var \Drupal\feeds_migrate_ui\FeedsMigrateUiParserSuggestion
-   */
-  protected $parserSuggestion;
-
-  /**
-   * Fill This.
-   *
-   * @var \Drupal\migrate_plus\AuthenticationPluginManager
-   */
-  protected $authPlugins;
-
-  /**
-   * Fill This.
-   *
-   * @var \Drupal\migrate_plus\DataFetcherPluginManager
-   */
-  protected $fetcherPlugins;
-
-  /**
-   * Fill This.
+   * Plugin manager for source plugins.
    *
-   * @var \Drupal\feeds_migrate\MappingFieldFormManager
+   * @var \Drupal\migrate\Plugin\MigratePluginManagerInterface
    */
-  protected $mappingFieldManager;
+  protected $sourcePluginManager;
 
   /**
-   * Fill This.
+   * Plugin manager for destination plugins.
    *
-   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   * @var \Drupal\migrate\Plugin\MigratePluginManagerInterface
    */
-  protected $entityTypeManager;
+  protected $destinationPluginManager;
 
   /**
-   * Fill This.
-   *
-   * @var \Drupal\Core\Entity\EntityFieldManager
-   */
-  protected $fieldManager;
-
-  /**
-   * Fill This.
+   * The form factory.
    *
-   * @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface
+   * @var \Drupal\feeds_migrate\Plugin\MigrateFormPluginFactory
    */
-  protected $bundleManager;
+  protected $formFactory;
 
   /**
-   * Fill This.
+   * The renderer service.
    *
-   * @var \Drupal\feeds_migrate\DataParserPluginManager
+   * @var \Drupal\Core\Render\Renderer
    */
-  protected $parserManager;
+  protected $renderer;
 
   /**
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container) {
     return new static(
       $container->get('plugin.manager.migration'),
-      $container->get('feeds_migrate.plugin_form_factory'),
-      $container->get('plugin.manager.feeds_migrate.data_parser_form'),
-      $container->get('feeds_migrate_ui.parser_suggestion'),
-      $container->get('plugin.manager.feeds_migrate.authentication_form'),
-      $container->get('plugin.manager.feeds_migrate.data_fetcher_form'),
-      $container->get('plugin.manager.feeds_migrate.mapping_field_form'),
-      $container->get('entity_field.manager')
+      $container->get('plugin.manager.migrate.source'),
```

Comment:
We no longer need the plugin managers for data fetcher and parser since those are always part of the source plugin configuration itself. They will be handled accordingly on the source plugin form.